### PR TITLE
Only generate the prisma client if it's not found

### PR DESCRIPTION
**Pull request by dillonraphael**
Closes: #3801 , https://github.com/blitz-js/blitz/issues/3776

### What are the changes and their implications?

Only generate the prisma client if it's not found. It checks for both `@prisma/client` and `.prisma`.

### CODE REVIEW

File: packages/blitz/src/cli/utils/codegen-tasks.ts [hunk 1]
```diff
@@ -39,33 +39,23 @@ export const codegenTasks = async () => {
...
+        if (result.success) {
+          prismaSpinner.succeed(log.greenText("Generated Prisma client"))
         } else {
-            log.success("Prisma client was up to date")
+          prismaSpinner.fail()
+          console.log("\n" + result.stderr)
+          process.exit(1)
         }
+      } else {
+        log.success("Prisma client was up to date")
```
Comment (beerose) [inline packages/blitz/src/cli/utils/codegen-tasks.ts:57]: I'm not sure about this message. According to the code, prisma client **was found**, and not necessarily **up to date**. Saying "up to date" can be misleading.

Comment (beerose) [reply packages/blitz/src/cli/utils/codegen-tasks.ts:57]: If prisma client was found, then we're skipping the `prisma generate` step. IMO we shouldn't print anything in this case.

File: packages/blitz/src/cli/utils/codegen-tasks.ts [hunk 2]
```diff
@@ -39,33 +39,23 @@ export const codegenTasks = async () => {
...
-                console.log("\n" + result.stderr)
-                process.exit(1)
-            }
-            await fs.writeJson(join(process.cwd(), `blitz-${creationDate.birthtimeMs}-prisma-client.json`), {
-                lastModified: lastModified.mtimeMs,
-            })
+      const foundPrismaClient = resolveFrom.silent(process.cwd(), "@prisma/client")
+      if (
+        !foundPrismaClient ||
+        (foundPrismaClient && !fs.existsSync(join(foundPrismaClient, "../../..", ".prisma")))
```
Comment (beerose) [inline packages/blitz/src/cli/utils/codegen-tasks.ts:45]: Does it work for both pnpm and not-pnpm package managers?

**Review by beerose — COMMENTED** [review]
Are we sure it works in monorepos? 

From what I saw, it finds a prisma client in the monorepo's root `node_modules`, while I guess it should look for a particular app's `node_modules`.

File: packages/blitz/src/cli/utils/codegen-tasks.ts (hunk 2 above)
Comment (dillonraphael) [reply packages/blitz/src/cli/utils/codegen-tasks.ts]: Just pushed a fix. Tested with flightcontrol as well to get it working.

File: packages/blitz/src/cli/utils/codegen-tasks.ts (hunk 1 above)
Comment (dillonraphael) [reply packages/blitz/src/cli/utils/codegen-tasks.ts]: Ya very valid point

File: packages/blitz/src/utils/run-prisma.ts [hunk 3]
```diff
@@ -1,27 +1,32 @@
...
 
-  return {
-    success: code === 0,
-    stderr: silent ? cp_stderr.join("") : undefined,
+    return {
+      success: code === 0,
+      stderr: silent ? cp_stderr.join("") : undefined,
+    }
+  } else {
+    log.error("Prisma bin not found")
```
Comment (beerose) [inline packages/blitz/src/utils/run-prisma.ts:30]: It should return in each case, as other usages expect this function to return a value, e.g.:
CleanShot 2022-09-23 at 15 44 04@2x

Not sure why CI didn't catch this.

**Review by beerose — CHANGES_REQUESTED** [review]

**Review by beerose — APPROVED** [review]

**Pull request merged**